### PR TITLE
loadbalancer additional_vips by snigle

### PR DESCRIPTION
**Pull request by EmilienM**
Initially started with https://github.com/gophercloud/gophercloud/pull/2700 by @snigle - but we have tried to reach the author with no success, and some folks needs this PR.
To get it, we just need it rebased.

Fixes https://github.com/gophercloud/gophercloud/issues/2699

https://docs.openstack.org/api-ref/load-balancer/v2/?expanded=show-load-balancer-details-detail#show-load-balancer-details

### CODE REVIEW

File: internal/acceptance/openstack/loadbalancer/v2/loadbalancers_test.go [hunk 1]
```diff
@@ -262,18 +262,29 @@ func TestLoadbalancersCRUD(t *testing.T) {
...
+	th.AssertNoErr(t, err)
+	defer networking.DeletePort(t, netClient, additionalSubnetPort.ID)
+
 	policy1, err := policies.CreateQoSPolicy(t, netClient)
 	th.AssertNoErr(t, err)
 	defer policies.DeleteQoSPolicy(t, netClient, policy1.ID)
 
 	tags := []string{"test"}
-	lb, err := CreateLoadBalancer(t, lbClient, subnet.ID, tags, policy1.ID)
+	lb, err := CreateLoadBalancer(t, lbClient, subnet.ID, tags, policy1.ID, []loadbalancers.AdditionalVip{{SubnetID: additionalSubnet.ID, IPAddress: additionalSubnetPort.FixedIPs[0].IPAddress}})
```
Comment (EmilienM) [inline internal/acceptance/openstack/loadbalancer/v2/loadbalancers_test.go:282]: e2e fails here:

```
2024-01-16T14:08:43.5681807Z     loadbalancer.go:117: Attempting to create loadbalancer TESTACCT-d8omS1Ki on subnet 9a805cbc-52c8-4ba2-8e23-cd0dbcb67779
2024-01-16T14:08:43.9458298Z     convenience.go:36: [1;31mFailure in loadbalancers_test.go, line 283: unexpected error [0m[1;33m"Internal Server Error"[0m[1;31m[0m
```

Comment (EmilienM) [reply internal/acceptance/openstack/loadbalancer/v2/loadbalancers_test.go:282]:
```
ERROR octavia.network.drivers.neutron.allowed_address_pairs openstack.exceptions.ConflictException: ConflictException: 409: Client Error for url: http://10.1.0.222:9696/networking/v2.0/ports, IP address 192.168.2.207 already allocated in subnet c210afcf-a265-467d-a844-25aa7e981dd7
```

**Comment by EmilienM** [comment]
@pierreprinetti any chance to look at it when time permits?
Thanks

**Review by mandre — APPROVED** [review]
LGTM

**Pull request merged**